File: JustMathsJavaApp/app/src/main/java/com/example/justmathsjavaapp/MainActivity.java
package com.example.justmathsjavaapp;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.ImageButton;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {
    ImageButton btnMathPlay, btnMathShare, btnMathRate;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        btnMathPlay = findViewById(R.id.btnMathPlay);
        btnMathShare = findViewById(R.id.btnMathShare);
        btnMathRate = findViewById(R.id.btnMathRate);

        btnMathPlay.setOnClickListener(v -> {
            /*
             *  Example of Explicit Intent
             *  When you click Play Button on the screen
             *  Game Activity will be started
             */

            Intent i = new Intent(MainActivity.this, GameActivity.class);
            startActivity(i);
        });

        btnMathShare.setOnClickListener(v -> {
            /*
             *  Example of Implict Intent
             *  When you click Share Button on the screen
             *  Android will find the activities that allow to share messages.
             */

            Intent intent = new Intent(Intent.ACTION_SEND);
            intent.setType("text/plain");
            intent.putExtra(Intent.EXTRA_TEXT, "Just Maths - Fun way to learn Maths. http://www.play.google.com");
            startActivity(intent);
        });

        btnMathRate.setOnClickListener(v -> {
            /*
             *  Simple Toast Message To Display Message For Short Duration
             *  Link that to your app landing page.
             */
            Toast.makeText(MainActivity.this, "You can open your Google Play landing page", Toast.LENGTH_LONG).show();
        });
    }
}

File: LoginApp/app/src/main/java/com/example/loginapp/MainActivity.kt
package com.example.loginapp

import android.content.Intent
import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.google.firebase.auth.FirebaseAuth
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {
    private lateinit var auth: FirebaseAuth

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // Initialize Firebase Auth
        auth = FirebaseAuth.getInstance()

        // Register
        btnRegister.setOnClickListener {
            val intent = Intent(this, RegisterActivity::class.java)
            startActivity(intent)
            finish()
        }

        // Login
        btnLogin.setOnClickListener {
            if (Email.text.toString().trim { it <= ' ' }.isNotEmpty() &&
                Password.text.toString().trim { it <= ' ' }.isNotEmpty()
            ) {
                val email = Email.text.toString()
                val pwd = Password.text.toString()
                auth.signInWithEmailAndPassword(email, pwd)
                    .addOnCompleteListener(this) {
                        if (it.isSuccessful) {
                            val intent = Intent(this, LoggedIn::class.java)
                            intent.putExtra("email", email)
                            startActivity(intent)
                            finish()
                        } else {
                            Toast.makeText(this, "Wrong Details", Toast.LENGTH_LONG).show()
                        }
                    }
            } else {
                Toast.makeText(this, "Enter the Details", Toast.LENGTH_LONG).show()
            }
        }
    }
}